### PR TITLE
new(tests): Exported Evmone EOF unit tests

### PR DESCRIPTION
**Pull request by hugo-dc**
## 🗒️ Description

- **Includes python tests exported from evmone's unit tests**
- Adds missing EOF Exceptions
- Consider if the tested bytecode is initcode or runtime when checking for duplicates

## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_eof1_invalid_type_section_size.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+              ,
+              "ef00010100100200030001000100010400000000800000008000000080000000800000fefefe",
+              EOFException.INVALID_TYPE_SECTION_SIZE,
+              id="eof1_invalid_type_section_size_4"
+        ),
+        
+    ]
+)
+
+def test_example_valid_invalid(
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_eof1_invalid_type_section_size.py:169]: Very nice test scenarios.

Similar to previous comment, we could:
- Rename file to `test_type_section_size.py`
- Rename test function to `test_invalid_containers`
- Improve the docstring description

In general I think the `test_eof1` prefix for file names in this folder is redundant since the version of EOF containers we are testing here can be implied from the parent folders' names.

**Review by marioevz — COMMENTED** [review]
Very nice work overall!

I haven't finished the review but I thought of sharing the comments I have so far in order to start the rework earlier.

Overall general comments:
- We should reduce the number of files, since the generated test case names already contain the test file name and the test function name, many of the test cases could be placed in existing files (primarily `tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py`) without risking name collisions.
- On the new files, we should skip the `test_eof1_` prefix and instead simply use `test_`.
- Linter is currently complaining about the format of many of the files. One simple solution is to run the `black` python formatting tool (see `uv run black --help`) on the new files.

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_unreachable_code_sections.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+                    )
+                ),
+            ],
+            expected_bytecode="ef0001010004020001000b030001003804000000008000046000600060006000ec0000ef000101001402000500030003000100030003040000000080000000800000008000000080000000800000e50001e50002fee50004e50003",
+            validity_error=[EOFException.UNREACHABLE_CODE_SECTIONS],
+        ),
+    ],
+    ids=lambda c: c.name,
+)
+def test_unreachable_code_sections(
```
Comment (danceratopz) [inline tests/osaka/eip7692_eof_v1/eip4750_functions/test_unreachable_code_sections.py:312]: This function is causing the fails in the "Build html documentation..." CI Job https://github.com/ethereum/execution-spec-tests/actions/runs/13830246793/job/38692699658

TLDR: This module doesn't exist any longer in `main` and this module your branch looks to be out of sync, whereas the EOF Tracker reflects the expected state from `main`. So the EOF Tracker is trying to include a link to a test that doesn't exist in this branch.

More details...

It looks like this python module (`tests/osaka/eip7692_eof_v1/eip4750_functions/test_unreachable_code_sections.py`) got renamed in `main` to `tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py` in #869. See the current state in `main` here:
https://github.com/ethereum/execution-spec-tests/blob/d4b536d466a1671f309c201efa71fe3ebed4f8ed/tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py#L485

The EOF Tracker in this branch has been updated to reflect the state in `main`:
https://github.com/ethereum/execution-spec-tests/blob/4fdc1d777d29a884813ab8e4cfc0f60a33afb89e/tests/osaka/eip7692_eof_v1/eof_tracker.md?plain=1#L155-L158
